boot/kernel_os.go

```diff
@@ -61,7 +61,13 @@ func ExtractKernelAssets(s *snap.Info, snapf snap.Container) error {
 		return fmt.Errorf("cannot extract kernel assets: %s", err)
 	}
 
-	if bootloader.Name() == "grub" {
+	// XXX: should we use "kernel.yaml" for this?
+	var forceKernelExtraction bool
+	if _, err := snapf.ReadFile("force-kernel-extraction"); err == nil {
```

Comment:
Which snap would be authoritative to know about this? Is the kernel or gadget snap the entity that should have this voice?

Reply:
When we discussed this last our thinking was that its the kernel because the (only) use-case we have so far is the full-disk encryption. And that needs a matching initramfs change so its the kernel that knows that it needs this kind of extraction.

Reply:
Should we have a structure like `meta/kernel/requires-extraction` that can be easily checked from any environment, similar to how experimental features are present as simple empty files?

The way I'm reading the code now would suggest that the file has to be present in the root of the file system of the snap, which feels ugly.

Reply:
We don't want to make a pattern of this,  if we ought to we would introduce kernel.yaml.

I agree though that having it live under "meta/force-kernel-extraction"  seems more sensible.